lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -303,7 +304,12 @@ fCalibMJ := SEL(fFrequency <> 0, -9999, fPulseWattage / fFrequency);
         uHardWavelengthEdge3 .. 65535: bOverAllowableEnergy := fCalibMJ > 10;
     END_CASE
 {END_IF}
-FFO(i_xOK := NOT bOverAllowableEnergy,
+IF eEnumGet <> E_PPM_States.OUT AND bOverAllowableEnergy = TRUE THEN
+    uOverCounter := MIN(uOverCounter + 2, 50);
+ELSE
+    uOverCounter := MAX(uOverCounter - 1, 1);
+END_IF
+FF(i_xOK := eEnumGet = E_PPM_States.OUT OR uOverCounter < 50,
```

Comment:
I'm trying to get into the habit of verbosely double-checking fast faults now:

Fault is OK if:
- PPM is out
- OR counter is under 50

Therefore, the beam turns off when both:
- PPM is at one of the known target or an unknown state or in motion
- AND the counter is at least 50 (counter caps out at 50)

This is totally reasonable logic and I endorse it

Some might argue that e.g. the YAG states should also be exempt from these faults, but there is some judgement call to be made because what if the beam is actually hitting the power meter somehow in these states? And what if the state positions other than "OUT" are wrong? I think there's a philosophical discussion to be had here.